source/use-cases/sensitive-data-encryption.txt

```diff
@@ -124,7 +124,153 @@ In this step, we generate a local master key to expedite setup of our developmen
           }
 
 
-B. Define a JSON Schema
+B. Create a Data Encryption Key
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this section, we use the MongoDB driver to create and store a data
+encryption key, used by CSFLE to automatically encrypt and decrypt data on
+document fields. The following configuration values are required by the client
+to generate a new data encryption key:
+
+* The local master key or AWS KMS master key access settings.
+* A MongoDB connection string that authenticates on a running server.
+* The key vault database and collection name.
+
+Upon creating a data encryption key, the driver returns the ``_id`` field
+(as a `UUID <https://en.wikipedia.org/wiki/Universally_unique_identifier>`_)
+of a document that stores information about the key including:
+
+* The data encryption key, encrypted by the master key.
+* The key management service provider data.
+
+Follow the steps below to generate a single data encryption key from the
+local master key.
+
+.. note::
+
+  If using AWS KMS to provide the master key, refer to the "`F. Convert to
+  a Remote Master Key`_" section for the appropriate settings.
+
+First, read the file containing the local master key generated in the "`A.
+Create a Master Key`_" section:
+
+.. tabs::
+
+  tabs:
+
+    - id: java-master-key-reader
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          String path = "master-key.txt";
+
+          byte[] fileBytes = new byte[96];
+
+          try (FileInputStream fis = new FileInputStream(path)) {
+            fileBytes = fis.readAllBytes();
+          } catch (IOException e) {
+            e.printStackTrace();
+          }
+
+          final byte[] localMasterKey = Arrays.copyOf(fileBytes, 96);
+
+
+Next, specify the KMS provider settings. Since we are using a local master
+key, the provider name is ``local``:
+
+.. tabs::
+
+  tabs:
+
+    - id: java-kms-settings-local
+      name: "Java"
+      content: |
+
+        .. code-block:: java
+
+          String kmsProvider = "local";
+
+          Map<String, Map<String, Object>> kmsProviders = new HashMap<String, Map<String, Object>>() {{
+             put(kmsProvider, new HashMap<String, Object>() {{
+                 put("key", localMasterKey);
+             }});
+          }};
+
+
+Create the client with the MongoDB connection string and key vault namespace
+(database and collection name):
```

Comment:
Let's give specific guidance about what the namespace should be. If we have a best practice, we should say it. Otherwise, we should explicitly say that the names are up to the developer.

Reply:
Would it be ok if I just make a note to the user that is the arbitrary namespace we will be using for now? I discussed with Kenn and Ravind yesterday and the conclusion was that they would work on something more standardized during "beta cleanup". Conversation in #fle on Slack starts here:
https://mongodb.slack.com/archives/CDM9QEGUT/p1567530749015300